scripts: sbom: Add GN external project support for Matter (#9937)
nrfconnect/sdk-nrf: scripts/west_commands/sbom/README.rst

  The :ref:`west_sbom Extracting from build` section describes in detail how to extract a list of files from a build directory.

  You can use ``-d`` option multiple times.

>   For example, to include both the ``mcuboot`` child image and the main application, you can use the following command:

Reviewer: ```suggestion
  For example, to include both the ``mcuboot`` child image and the main application, use the following command:
```

     :class: highlight

     -d build_directory *target.elf*:*file.map*

Integration with the GN meta-build system
=========================================

> The ``ncs-sbom`` reads a list of all commands needed to build provided targets.

Reviewer: ```suggestion
The ``ncs-sbom`` script reads a list of all commands needed to build provided targets.
```

> If there is a ``gn gen`` command, the script enters ``gn`` build directory and tries to extract files from it using the same method as described above.

Reviewer: ```suggestion
If there is a ``gn gen`` command, the script enters the :file:`gn` build directory and tries to extract files from it using the same method as described earlier.
```

Reviewer 2: I think it's supposed to be "the script enters the build directory used by GN and...". `gn`is not the name of the directory.

Author: Maybe it can be something like that?
```
If there is a ``gn gen`` command, the script enters the command's build
directory and tries to extract files from it using the same method
as described earlier.
```

Author: @peknis, what do you think about above sentence?
